pkg/providers/ingress/translation/annotations/upstreamscheme/upstreamscheme.go

```diff
@@ -0,0 +1,50 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package upstreamscheme
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apache/apisix-ingress-controller/pkg/providers/ingress/translation/annotations"
+)
+
+type upstreamscheme struct{}
+
+func NewParser() annotations.IngressAnnotationsParser {
+	return &upstreamscheme{}
+}
+
+var schemeMap map[string]int = map[string]int{
+	"http":  1,
```

Comment:
If the map values are useless, I prefer to use an empty struct value here. Otherwise, other readers may get confused.

Reply:
+1

Reply:
Indeed an `empty struct` is better
```go
map[string]struct{}
```